qhub/schema.py

```diff
@@ -364,6 +364,29 @@ class Main(Base):
     environments: typing.Dict[str, CondaEnvironment]
     clearml: typing.Optional[ClearML]
 
+    @validator("project_name")
+    def project_name_convention(cls, value: typing.Any):
+        convention = """
+        In order to successfully deploy QHub, there are some project naming conventions which need
+        to be followed. First, ensure your name is compatible with the specific one for
+        your chosen Cloud provider. In addition, the QHub project name should also obey the following
+        format requirements:
+        - letters from A to Z (upper and lower case) and numbers;
+        - Special characters are not allowed, such as "!@#$%^&*()";
+        - Maximum accepted length of the name string is 16 characters.
+        - If using AWS: names should not start with the string "aws";
+        """
+        if len(value) > 16:
+            raise ValueError(
+                "Maximum accepted length of the project name string is 16 characters."
+            )
+        elif ProviderEnum == AzureProvider and re.findall(
+            r"^[A-Za-z0-9][^/|.~!?@#$%^=&*\\()_]*[A-Za-z0-9]$", value
```

Comment:
I'd rather have a simple regex and rules for users on what can be in a project name so that we don't have to have so many exceptions. For example:
  - must begin with a letter `[a-z][A-Z]`
  - cannot have name begin with `aws`
  - name can only contain `[a-zA-Z0-9]`

Can you make this the rule for checking?  Is this sane?